### PR TITLE
Small fixes to json-c, zip, unzip and fontconfig

### PR DESCRIPTION
**Pull request by no92**

### CODE REVIEW

**Review by Dennisbonke — APPROVED** [review]
LGTM

File: patches/json-c/0001-Add-managarm-specific-changes.patch [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+diff --git a/json_tokener.c b/json_tokener.c
+index 0c09b66..d74153a 100644
+--- a/json_tokener.c
++++ b/json_tokener.c
+@@ -327,7 +327,7 @@ struct json_object *json_tokener_parse_ex(struct json_tokener *tok, const char *
+ 		if (newloc == NULL)
+ 		{
+ 			freelocale(duploc);
+-			return NULL;
++			// return NULL;
```
Comment (avdgrinten) [inline patches/json-c/0001-Add-managarm-specific-changes.patch:19]: Why is this done?

Comment (no92) [reply patches/json-c/0001-Add-managarm-specific-changes.patch:19]: Due to our stubbing of locale functions, this prevents json-c from parsing input properly.

Comment (avdgrinten) [reply patches/json-c/0001-Add-managarm-specific-changes.patch:19]: Is the resulting code path correct? I'd expect that there is some reason that they return and don't ignore the error here. IMHO such cases should receive comments (or this should be fixed in mlibc).

Comment (no92) [reply patches/json-c/0001-Add-managarm-specific-changes.patch:19]: The parsing after this is correct. The problem that this is running into is that mlibc stubs locale functions, and returns nullptrs. This mostly works fine, but json-c treats this as an error condition and stops parsing, which is not what we want here.

**Pull request merged**